libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -136,12 +136,14 @@ WaveClip::~WaveClip()
 {
 }
 
-AudioSegmentSampleView
-WaveClip::GetSampleView(size_t ii, sampleCount start, size_t length) const
-{
-   assert(ii < GetWidth());
-   return mSequences[ii]->GetFloatSampleView(
-      start + TimeToSamples(mTrimLeft), length);
+ChannelSampleViews
+WaveClip::GetSampleView(sampleCount start, size_t length) const
+{
+   ChannelSampleViews channels;
+   for (auto i = 0u; i < GetWidth(); ++i)
+      channels.push_back(mSequences[i]->GetFloatSampleView(
+         start + TimeToSamples(mTrimLeft), length));
```

Comment:
You are assuming here that all channels will be needed in the cache.  But remember what I explained about drawing code, including spectrogram, needing to stride "channel-major" but "clip-minor".

The caches for spectrogram drawing might now be doing wasteful duplicated work?  Creating a left and right side cache, but only using one side, destroying that cache, then repeating for the right channel?

.

Reply:
Oh, never mind.  The caching for spectrum view will not include the stretching decorator.

Reply:
Not the stretching decorator, but `CachingPlayableSequence` also uses `WaveClip::GetSampleView`.
Currently it calls it on two separate tracks each with mono clips, so there's no unnecessary work. When we do use wide clips, though, there'll be only one stereo track, `WaveTrack::GetSampleView` will call `GetSampleView` on clips, and yes, doing unnecessary work if it stays at that.
Looks like I indeed need to preserve the `ii` argument ...

Reply:
If you just make a `// TODO` note to revisit the efficiency problem in a later PR, I would accept that.

Reply:
I'm addressing it.